contracts/Auction.sol

```diff
@@ -96,16 +97,42 @@ contract Auction {
     }
 
 
-    function sendFunds(address x, address y, fixed128x10 cost) public {
+    function sendFunds(address x, address y) public {
 
     }
 
     /*
         Place a bid on the item. The bid is added to the history of bids list. Incoming bid must be higher than the highest bid otherwise reject the bid. If incoming bid
         is higher than highest bid, then the highest bidder's funds should be released back to he/she and update highest bid.
     */
-    function bid(address bidder, uint bid_amount) payable public {
-        bid_history.push(bid_record(bidder, bid_amount, block.timestamp));
+    function bid() payable public {
+        require(
+            block.timestamp <= end_time,
+            "Auction is over."
+        );
+        
+        require(
+            msg.value > current_highest_bid, 
+            // "The current highest bid is: ", current_highest_bid, ". You must place a higher bid"
+            "You must place a higher bid"
+        );
+        
+        /*
+            msg.sender (address payable): sender of the message (current call)
+            msg.value (uint): number of wei sent with the message
+        */
+        bid_history.push(bid_record(msg.sender, msg.value, block.timestamp));
+        
+        if (current_highest_bid == 0){
+            current_highest_bid = msg.value;
+            current_highest_bidder = msg.sender; 
+        }
+        else{
+            current_highest_bidder.transfer(address(this).balance);
```

Comment:
could you explain a bit this line, what's `address(this).balance` refer to, thanks !

Reply:
if you googled it then you would know what it does. just playing haha. it refers to the smart contract's ethereum balance